File: python/comext/comext_example.py
# Ejemplo de uso — Eurostat COMEXT (JSON -> DataFrame -> CSV)
# ----------------------------------------------------------------------------
# 1) Cargar la función
# 2) Ejecutar la consulta con filtros (parámetros repetidos para multiselección)
# 3) Guardar el resultado como CSV
# ----------------------------------------------------------------------------
import os
from comext_function import comext_api_function


def main() -> None:
    # Asegurar rutas relativas coherentes
    os.chdir(os.path.dirname(os.path.abspath(__file__)))

    # Ejecutar la consulta
    df = comext_api_function(
        dataset_id="DS-059341",
        filters={
            "reporter": ["ES"],
            "partner": ["US"],
            "product": ["1509"],
            "flow": ["2"],
            "freq": ["A"],
            "time": ["2019", "2020"],
        },
    )

    # Guardar el resultado
    df.to_csv("comext_example.csv", index=False)


if __name__ == "__main__":
    main()

File: R/comext/comext_example.R
# Ejemplo de uso de la función comext_api_function
library(httr)
library(jsonlite)

#setwd("comext")
source("comext_function.R")

# Ejemplo de uso
df <- comext_api_function(
  dataset_id = "DS-059341",
  filters = list(
    reporter = c("ES"),
    partner  = c("US"),
    product  = c("1509", "8703"),
    flow     = c("2"),
    freq     = c("A"),
    time     = 2015:2020
  )
)

write.csv(df, "comext_example.csv", row.names = FALSE)


